### ISSUE TITLE
NCDAP tests fail on Windows

### ISSUE THREAD

**Issue by WardF** [Unidata/netcdf-c#96]
After fixing the compile issues in https://github.com/Unidata/netcdf-c/issues/95, the following tests are failing on Windows:

* `ncdap_test_nstride_cached`
* `ncdap_t_misc`
* `ncdap_test_varm3`

Investigating.

**Comment by WardF** [comment]
I've confirmed that this issue wasn't present in `v4.3.3-rc2`; I'm working on narrowing down where this was introduced.

**Comment by WardF** [comment]
It appears that the bug was introduced at some point in the pull request for ncf-177.  I'm trying to figure out where exactly it was introduced.  Unfortunately it appears that the compilation issue may have been introduced at the same time.  Trying to sort out a script I can use to automate a git bisect run, but unfortunately this will probably spill over into tomorrow at least.

**Comment by qkoziol** [comment]
> On Jan 12, 2015, at 4:54 PM, Ward Fisher <notifications@github.com> wrote:
> 
> It appears that the bug was introduced at some point in the pull request for ncf-177. I'm trying to figure out where exactly it was introduced. Unfortunately it appears that the compilation issue may have been introduced at the same time. Trying to sort out a script I can use to automate a git bisect run, but unfortunately this will probably spill over into tomorrow at least.
> 

	Hmm, let me know if it's related to my changes and I'll try to figure out the problem.  But, it might be somewhat tricky for me, since I don't think I've got any easy access to a Windows development environment...

	Quincey=

**Comment by WardF** [comment]
Will do; it's not entirely clear yet, we'll see what git bisect has to say. Off hand I'd say probably not, since it's in an area of the code you didn't directly touch (I think), but the interleaving of merges and other interspersed issues means it will take a bit to pick through :)

**Comment by WardF** [comment]
I've been able to track down the compilation issue from issue #95 to commit f423f27, on December 24th.  Going to correct those in the detached head and see if the test issues were also introduced as well.

**Comment by WardF** [comment]
Tracked the issue down to two places in commit `f423f27`: First, there is an assertion that:

```C
sizeof(long) == sizeof(void*)
```
which does not hold true under 64-bit MSVC. Second, the following assertion in `ocinternal.c` fails:

```C
OCASSERT(state->curlflags.cookiejar != NULL);
```

This second issue is being caused by an error in `ocmktmp()` when a temporary file is created in Windows.  As a result, `state->curlflags.cookiejar` is NULL and this assertion fails. 

I'll correct these issues and see what else pops up.

**Comment by WardF** [comment]
Fixed in commit 393db0018e36bab41a4a19eb9f06ed9804021f8b.
